tests/test_uncrustify/test.py

```diff
@@ -253,7 +254,9 @@ def print_as_ctest(self, out_file=sys.stdout):
         def to_cmake_path(path):
             if type(path) is dict:
                 return {k: to_cmake_path(v) for k, v in path.items()}
-            return path.replace(os.sep, '/')
+            if type(path) is str:
```

Comment:
I take it this is what was "already broken"? What is the type of `path` if *not* dict or str? (I also wonder if this is correct for Python 2 vs. 3?)

Note: I'd be perfectly happy to drop this from this PR and follow up on it separately.

Reply:
yeah, it's broken because of `self.test_xfail = False` is not a string referring to a path.